adapters/openrtb_util_test.go

```diff
@@ -2,116 +2,100 @@ package adapters
 	}{
 		{
-			name: "extract bidder params from nil req",
-			args: args{
-				bidRequest: nil,
-			},
-			want:    nil,
-			wantErr: true,
+			name:            "nil req",
+			givenBidRequest: nil,
+			want:            nil,
+			wantErr:         errors.New("error bidRequest should not be nil"),
 		},
 		{
-			name: "extract bidder params from nil req.Ext",
-			args: args{
-				bidRequest: &openrtb2.BidRequest{Ext: json.RawMessage(`{"prebid":{}}`)},
-			},
-			want:    nil,
-			wantErr: false,
+			name:            "nil req.ext",
+			givenBidRequest: &openrtb2.BidRequest{Ext: json.RawMessage(`{"prebid":{}}`)},
+			want:            nil,
+			wantErr:         nil,
 		},
 		{
-			name: "extract bidder params from req.Ext for input request in adapter code",
-			args: args{
-				bidRequest: &openrtb2.BidRequest{Ext: json.RawMessage(`{"prebid":{"bidderparams": {"profile": 1234, "version": 1}}}`)},
-			},
-			want:    map[string]json.RawMessage{"profile": json.RawMessage(`1234`), "version": json.RawMessage(`1`)},
-			wantErr: false,
+			name:            "malformed req.ext",
+			givenBidRequest: &openrtb2.BidRequest{Ext: json.RawMessage("malformed")},
+			want:            nil,
+			wantErr:         errors.New("error decoding Request.ext : invalid character 'm' looking for beginning of value"),
+		},
+		{
+			name:            "extract bidder params from req.Ext for input request in adapter code",
+			givenBidRequest: &openrtb2.BidRequest{Ext: json.RawMessage(`{"prebid":{"bidderparams": {"profile": 1234, "version": 1}}}`)},
+			want:            map[string]json.RawMessage{"profile": json.RawMessage(`1234`), "version": json.RawMessage(`1`)},
+			wantErr:         nil,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := ExtractAdapterReqBidderParams(tt.args.bidRequest)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("ExtractReqExtBidderParams() error = %v, wantErr = %v", err, tt.wantErr)
-				return
-			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("ExtractReqExtBidderParams() got = %v, want = %v", got, tt.want)
-			}
+			got, err := ExtractReqExtBidderParamsMap(tt.givenBidRequest)
+			assert.Equal(t, tt.wantErr, err, "err")
+			assert.Equal(t, tt.want, got, "result")
 		})
 	}
 }
 
-func TestExtractReqExtBidderParams(t *testing.T) {
-	type args struct {
-		request *openrtb2.BidRequest
-	}
+func TestExtractReqExtBidderParamsMap(t *testing.T) {
 	tests := []struct {
-		name    string
-		args    args
-		want    map[string]map[string]json.RawMessage
-		wantErr bool
+		name            string
+		givenBidRequest *openrtb2.BidRequest
+		want            map[string]map[string]json.RawMessage
+		wantErr         error
 	}{
 		{
-			name: "extract bidder params from nil req",
-			args: args{
-				request: nil,
-			},
-			want:    nil,
-			wantErr: true,
+			name:            "nil req",
+			givenBidRequest: nil,
+			want:            nil,
+			wantErr:         errors.New("error bidRequest should not be nil"),
```

Comment:
It's better to compare the actual error message rather than verify the count of errors.